### DISCUSSION TITLE
1.21 file structure and json files

### DISCUSSION THREAD

**Discussion by juice-ix** [IoeCmcomc/MCDatapacker#23] [General]
I figured that changing the entire syntax to support 1.21 would be difficult, given how they changed the whole system for items, but it would be nice if we could get a beta that uses the file structure, and misode's generators for 1.21,.

**Comment by IoeCmcomc** [comment]
In my opinion, implementing the 1.21 file structure shouldn't be too hard, although I haven't touched this part and item components yet. My current priority is supporting 1.21 command syntaxes, then adding 1.21 information of blocks, items, and so on, and then updating side editors.

I don't release beta versions because I don't know whether someone wants to test these versions, but now I'll reconsider about beta versions.

Adding support for [Misode's generators](https://misode.github.io/) is something that has been on my mind for a while. If succeeded, it would reduce my work on the program a lot. However, Misode's generators are written as a web application, using HTML, CSS, and TypeScript, which normally requires a browser or similar environments to operate. In contrast, MCDatapacker is written as a C++ application with the Qt framework. Fortunately, Qt seems to have support for [integrating web content into Qt applications](https://doc.qt.io/qt-5/topics-web-content.html). Nevertheless, the idea is still fairly complicated, and I haven't experimented with it yet.

**Reply by juice-ix** [reply]
What I meant by the misode generator was the built in ones, i just assumed you had some sort of access to them so they would update as misode updated the generators. So do you make them yourself manually or how does that work?

**Reply by IoeCmcomc** [reply]
The built-in "generators" (I call them "side editors") are implemented manually and independently. Therefore, it will take me a while to update these. Misode's generators are just the main inspiration for the feature.
